presto-main/src/test/java/com/facebook/presto/sql/planner/TestEffectivePredicateExtractor.java

```diff
@@ -86,7 +86,11 @@
 @Test(singleThreaded = true)
 public class TestEffectivePredicateExtractor
 {
-    private static final TableHandle DUAL_TABLE_HANDLE = new TableHandle(new ConnectorId("test"), new TestingTableHandle());
+    private static final TableHandle DUAL_TABLE_HANDLE = new TableHandle(
+            new ConnectorId("test"),
+            new TestingTableHandle(),
+            TestingTransactionHandle.create(),
+            Optional.of(TestingHandle.INSTANCE));
```

Comment:
This one is interesting, since the `TableScanNode` create by using `DUAL_TABLE_HANDLE` can both use `Optional.empty()` and `Optional.of(TESTING_TABLE_LAYOUT)` as the `TableLayoutHandle`.